### ISSUE TITLE
Update documentation for type_name attribute on aws_kinesis_firehose_delivery_stream resource

### ISSUE THREAD

**Issue by clebio** [terraform-providers/terraform-provider-aws#15036]
The documentation for `kinesis_firehose_delivery_stream` states that `type_name` is required:

> type_name - (Required) The Elasticsearch type name with maximum length of 100 characters.

This is no longer accurate.

[1]: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kinesis_firehose_delivery_stream#type_name

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
$ terraform version
Terraform v0.12.28
+ provider.aws v3.5.0
```
### Affected Resource(s)

* kinesis_firehose_delivery_stream

### Terraform Configuration Files

```hcl

resource "aws_kinesis_firehose_delivery_stream" "waf" {
  name        = "aws-waf-logs-es"
  destination = "elasticsearch"
  # ...

  elasticsearch_configuration {
     # ...
   type_name  = "waf" 
  }
}
```

### Debug Output
n/a

### Panic Output
n/a

### Expected Behavior
plan/apply doesn't fail

### Actual Behavior
```
Error: error creating Kinesis Firehose Delivery Stream: InvalidArgumentException: Types are deprecated in Elasticsearch version 7.7. TypeName must be empty.
```

### Steps to Reproduce

1. `terraform apply -auto-approve`

### Important Factoids
n/a

### References

* #13032

**Comment by bflad** [comment]
Hi @clebio 👋  Thank you for reporting this and sorry you ran into trouble here. This does appear to be outdated documentation. The current resource schema does denote this as optional instead of required:

https://github.com/terraform-providers/terraform-provider-aws/blob/374a49a5878e95a70e7c2a4f3cb88a6e68da8df0/aws/resource_aws_kinesis_firehose_delivery_stream.go#L1218-L1222

If you or anyone is interested in updating the documentation for this argument, it can be found in the [`website/docs/r/kinesis_firehose_delivery_stream.html.markdown` file in this repository](https://github.com/terraform-providers/terraform-provider-aws/blob/master/website/docs/r/kinesis_firehose_delivery_stream.html.markdown). 👍

**Comment by anGie44** [comment]
The documentation update has been merged and will be reflected in `v3.8.0` of the provider, likely out this Thursday.
